libbeat/_meta/fields.ecs.yml

```diff
@@ -940,7 +940,46 @@
     
             This id normally changes across restarts, but `service.id` does not.
           example: 8a4f500f
-    
+
+    - name: source
+      title: Source
+      description: >
+        Source fields describe details about the source of the event.
+
+      type: group
+      fields:
+      - name: ip
+        level: core
+        type: ip
+        description: >
+          IP address of the source.
+
+          Can be one or multiple IPv4 or IPv6 addresses.
+
+      - name: hostname
```

Comment:
@andrewkroh @webmat This does not exist anymore in ECS. Do we need it for suricata? Can you two sync up?

Reply:
The ECS fields I used are stale because I moved this `source` definition from filebeat to libbeat, rather than getting an entirely new copy from elastic/ecs. We should do a full update in a separate PR (we have to verify that any fields removed from ECS are still accounted for elsewhere in fields.yml files when we do this).

Auditbeat had `source.hostname` in its fields.yml, so I had to remove it because it was now duplicated. I think Auditbeat uses `source.hostname` in cases where the SSH daemon reports both the remote client's IP and the reverse DNS name of that IP.

We can proceed with a rename in Auditbeat or a change ECS. Similarly we are using `source.geo.*` which is not in ECS.

Reply:
Just to clarify, `geo` is a reuseable object, and `source` and `destination` are both expected places where `geo` can be nested. The `template.json` doesn't reflect it yet, but the official spec is the readme :-)

So `source.geo.*` is perfectly ECS compliant.

Reply:
Agreed that `source.hostname` should be replaced by `source.domain`. Same goes for `destination.hostname`.

If you want to store an ambiguous value as well, before determining if it's a hostname or an IP, ECS no longer has a field for this (at least for now). You may store it in a custom field, or you may add logic to save to `.ip` only when it's an IP, or save to `.domain` when it's a hostname/domain.

Reply:
Ok, I'll update Auditbeat to send `source.domain` instead of `source.hostname`.

Reply:
I removed the `source.hostname` and `destination.hostname` fields. And I added the `source.geo.*` and `destination.geo.*`.

I reverted the changes to Auditbeat's mapping w.r.t. `source.hostname` as I will do that in an independent PR since it's separate change.